Ensuring data restrictions and pre-processing data availiability in archive.

EdwardColon-NOAA opened the pull request:
Ensuring that prepbufr and lightning files are save as restricted data as well as adding raw cloud, lightning and radar files needed for prep-processing in the hpss archive.

EdwardColon-NOAA committed:
Ensuring that prepbufr and lightning files are save as restricted data as well as adding raw cloud, lightning and radar files needed for prep-processing in the hpss archive.

EdwardColon-NOAA committed:
Ensuring that prepbufr and lightning files are save as restricted data as well as adding raw cloud, lightning and radar files needed for prep-processing in the hpss archive.

GangZhao-NOAA commented on scripts/akrtma3d/exakrtma3d_gsianl.ksh:
```diff
@@ -92,6 +92,8 @@ RUN_GUST="FALSE"
 # Link to the prepbufr data
 if [ -r ${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 ]; then
+   chgrp rstprod ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
+   chmod 750 ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
```
Hi @EdwardColon-NOAA 

Since these data files are not executable, I suggestion to use "chmod 640"  for here, also for the restricted data files in the following scripts. Also I checked the files under "obsproc" on wcoss2:
for rap prepbufr, under "/lfs/h1/ops/prod/com/obsproc/v1.2/rap.20250506",
-rw-r----- 1 ops.prod rstprod 34026872 May  6 11:29 rap.t11z.prepbufr.tm00

and for urma prepbufr under "/lfs/h1/ops/prod/com/urma/v2.10/urma2p5.20250506",
-rw-r----- 1 ops.prod rstprod 60517352 May  6 06:40 urma2p5.t00z.prepbufr
-rw-r----- 1 ops.prod rstprod 60378872 May  6 07:40 urma2p5.t01z.prepbufr
-rw-r----- 1 ops.prod rstprod 59567904 May  6 08:40 urma2p5.t02z.prepbufr
-rw-r----- 1 ops.prod rstprod 59241080 May  6 09:41 urma2p5.t03z.prepbufr
-rw-r----- 1 ops.prod rstprod 58751648 May  6 10:40 urma2p5.t04z.prepbufr

"chmod 750" still works, it would be "-rw**x**r-**x**---" with "**_x_**" in the file mode.  So my comment is only a suggestion. Basically I approved this PR.

Thank you!

-Gang

AnnetteGibbs-NOAA replied:
I agree with Gang's suggestion.  Also, have you tested all these changes?  Could you point us to the parallel running with these changes?  Shouldn't the file be called "${NET}.t${cyc}z.prepbufr.tm00" instead of ${NET}.t${HH}z.prepbufr.tm00?

AnnetteGibbs-NOAA commented on scripts/rtma3d/exrtma3d_gsianl.ksh:
```diff
@@ -86,6 +86,8 @@ RUN_GUST="FALSE"
 
 # Link to the prepbufr data
 if [ -r ${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 ]; then
+   chgrp rstprod ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
```
I think these files should be 

${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 not 
${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00

AnnetteGibbs-NOAA commented on scripts/rtma3d/exrtma3d_obsprep_lghtn.ksh:
```diff
@@ -136,6 +136,12 @@ export err=$?; err_chk
 
 msg="JOB $job FOR $NET HAS COMPLETED NORMALLY"
 postmsg "$jlogfile" "$msg"
+chgrp rstprod ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d
+chmod 750 ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d
+cpreq ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d ${COMINobsproc_rtma3d}
+cpreq ${DATA}/lightning_bufr.namelist ${COMINobsproc_rtma3d}
+chgrp rstprod ${DATA}/LightningInGSI.bufr
+chmod 750 ${DATA}/LightningInGSI.bufr
```
Don't you need a "cpreq -p" to preserve the permissions?

AnnetteGibbs-NOAA commented on scripts/rtma3d/exrtma3d_obsprep_radar.ksh:
```diff
@@ -123,6 +123,7 @@ if [ -s filelist_mrms ]; then
    mv filelist_mrms filelist_mrms_org
    ls MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 > filelist_mrms
    numgrib2=`more filelist_mrms | wc -l`
+   cpreq MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 ${COMINobsproc_rtma3d}
```
Shouldn't we copy the gzip'd files?  The files would be smaller and that is what the script is looking for if we do a rerun.

AnnetteGibbs-NOAA commented on scripts/akrtma3d/exakrtma3d_obsprep_lghtn.ksh:
```diff
@@ -136,6 +136,13 @@ export err=$?; err_chk
+
+chgrp rstprod ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d
+chmod 750 ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d
+cpreq ${DATA}/${NET}.t${cyc}z.lghtng.tm00.bufr_d ${COMINobsproc_rtma3d}
```
Don't you need a "cpreq -p" to preserve the permissions?

AnnetteGibbs-NOAA commented on scripts/akrtma3d/exakrtma3d_obsprep_radar.ksh:
```diff
@@ -123,6 +123,7 @@ if [ -s filelist_mrms ]; then
    mv filelist_mrms filelist_mrms_org
    ls MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 > filelist_mrms
    numgrib2=`more filelist_mrms | wc -l`
+   cpreq MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 ${COMINobsproc_rtma3d}
```
Shouldn't we copy the gzip'd files? The files would be smaller and that is what the script is looking for if we do a rerun.

AnnetteGibbs-NOAA reviewed (changes requested):
I made some suggestions and also found some errors.  Could you address the errors and retest.  Also, can you point us to your parallel that is testing these changes?  Thanks.

EdwardColon-NOAA committed:
Update exrtma3d_gsianl.ksh

EdwardColon-NOAA committed:
Update exakrtma3d_gsianl.ksh

EdwardColon-NOAA committed:
Update exakrtma3d_obsprep_lghtn.ksh

EdwardColon-NOAA committed:
Update exrtma3d_obsprep_lghtn.ksh

MatthewMorris-NOAA commented on scripts/akrtma3d/exakrtma3d_gsianl.ksh:
```diff
@@ -92,6 +92,8 @@ RUN_GUST="FALSE"
 
 # Link to the prepbufr data
 if [ -r ${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 ]; then
+   chgrp rstprod ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
```
These both need to be cyc instead of HH.

EdwardColon-NOAA commented:
Thanks for the input. I am making a few additional changes and testing.

-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566


On Tue, May 6, 2025 at 1:19 PM MatthewMorris-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/akrtma3d/exakrtma3d_gsianl.ksh
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#discussion_r2075938267>:
>
> > @@ -92,6 +92,8 @@ RUN_GUST="FALSE"
>
>  # Link to the prepbufr data
>  if [ -r ${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 ]; then
> +   chgrp rstprod ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
>
> These both need to be cyc instead of HH.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#pullrequestreview-2819080049>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQCQNSBPUZTSTXER3L325DVITAVCNFSM6AAAAAB4QDOY6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMJZGA4DAMBUHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

EdwardColon-NOAA commented:
The new changes to this PR are completed. The last check involves making
sure that the restricted files are tarred with their permissions intact.
Unfortunately, HPSS is down so I will not be able to verify until later
this evening.

-Edward

On Tue, May 6, 2025 at 1:28 PM Edward Colon - NOAA Affiliate <
***@***.***> wrote:

> Thanks for the input. I am making a few additional changes and testing.
>
> -------------------------------------------------
> Edward Colón
> *Lynker* at NOAA/NWS/NCEP/EMC
> 5830 University Research Ct.
> Rm #2025
> Riverdale, MD 20737
> Office: (301) 683-3815
> Cell: (301) 213-3566
>
>
> On Tue, May 6, 2025 at 1:19 PM MatthewMorris-NOAA <
> ***@***.***> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In scripts/akrtma3d/exakrtma3d_gsianl.ksh
>> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#discussion_r2075938267>:
>>
>> > @@ -92,6 +92,8 @@ RUN_GUST="FALSE"
>>
>>  # Link to the prepbufr data
>>  if [ -r ${OBS_DIR}/${NET}.t${cyc}z.prepbufr.tm00 ]; then
>> +   chgrp rstprod ${OBS_DIR}/${NET}.t${HH}z.prepbufr.tm00
>>
>> These both need to be cyc instead of HH.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#pullrequestreview-2819080049>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACFXXQCQNSBPUZTSTXER3L325DVITAVCNFSM6AAAAAB4QDOY6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMJZGA4DAMBUHE>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

EdwardColon-NOAA committed:
Update exrtma3d_gsianl.ksh

Removed chrgrp and chmod statements - not needed.

EdwardColon-NOAA committed:
Update exakrtma3d_gsianl.ksh

Removed chgrp and chmod statements - not needed.

EdwardColon-NOAA committed:
Update exakrtma3d_obsprep_lghtn.ksh

cpreq -p added to preserve restricted file status. Permissions modified for output bufr file.

EdwardColon-NOAA committed:
Update exakrtma3d_obsprep_radar.ksh

Saving gzipped rather than unzipped mrms grib2 files for archiving.

EdwardColon-NOAA committed:
Update exrtma3d_obsprep_lghtn.ksh

EdwardColon-NOAA committed:
Update exrtma3d_obsprep_radar.ksh

EdwardColon-NOAA committed:
Update exrtma3d_prepobs.ksh

EdwardColon-NOAA commented:
The gsianl ex-scripts were returned back to their original forms as modifications there were not needed. The changes were made in the lightning and radar exscripts as well as a single line change in prepobs which would be transferred to the gsianl exscript once prepobs is eliminated. 

The runhistory lists the newly added files and the restricted file check now tests positive for restricted files in the appropriate tarballs. 

/lfs/h2/emc/ptmp/edward.colon/RTMA3D_logs/conus/rtma3d_pbspro_rhist_YYYYMMDDHH00.log starting at 00Z for 20250507.

and

/lfs/h2/emc/ptmp/edward.colon/RTMA3D_logs/alaska/akrtma3d_pbspro_rhist_YYYYMMDDHH00.log starting at 00Z for 20250507.

AnnetteGibbs-NOAA commented on scripts/akrtma3d/exakrtma3d_obsprep_radar.ksh:
```diff
@@ -123,6 +123,7 @@ if [ -s filelist_mrms ]; then
    mv filelist_mrms filelist_mrms_org
    ls MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 > filelist_mrms
    numgrib2=`more filelist_mrms | wc -l`
+   cpreq ${COMINradar}/${obsname}/${obsname}_*_${YYYY}${MM}${DD}-${HH}${min}*.grib2.gz ${COMINobsproc_rtma3d}
```
These are copying the wrong files for some reason.  If you look in your obsprd.* directories, they are copying the 2 minute files instead instead of the 0 minute files.

AnnetteGibbs-NOAA commented on scripts/rtma3d/exrtma3d_obsprep_radar.ksh:
```diff
@@ -123,6 +123,7 @@ if [ -s filelist_mrms ]; then
    mv filelist_mrms filelist_mrms_org
    ls MergedReflectivityQC_*_${YYYY}${MM}${DD}-${HH}????.grib2 > filelist_mrms
    numgrib2=`more filelist_mrms | wc -l`
+   cpreq ${COMINradar}/${obsname}/${obsname}_*_${YYYY}${MM}${DD}-${HH}${min}*.grib2.gz ${COMINobsproc_rtma3d}
```
These are copying the wrong files for some reason. If you look in your obsprd.* directories, they are copying the 2 minute files instead instead of the 0 minute files (which are the files that are used).

AnnetteGibbs-NOAA reviewed (commented):
Hi Ed, the changes look good.  But for some reason the 2 minute MergedReflectivityQC files are being copied to the obsprd* directories instead of the 0 minutes files.  The 0 minutes files are actually used in the radar runs (compare the files in the obsprd* directories with the files in the data (obsprep_radar*) directories.  Can you fix this issue?

EdwardColon-NOAA commented:
Thanks Annette,

I am away from my desk but will update the PR as soon as I can.

-Edward

-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566


On Wed, May 7, 2025 at 9:57 AM Annette Gibbs ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Hi Ed, the changes look good. But for some reason the 2 minute
> MergedReflectivityQC files are being copied to the obsprd* directories
> instead of the 0 minutes files. The 0 minutes files are actually used in
> the radar runs (compare the files in the obsprd* directories with the files
> in the data (obsprep_radar*) directories.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#pullrequestreview-2821871036>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQCQ5ZVK55YSQBZXE2T25IGMHAVCNFSM6AAAAAB4QDOY6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMRRHA3TCMBTGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

MatthewMorris-NOAA commented on scripts/akrtma3d/exakrtma3d_obsprep_lghtn.ksh:
```diff
@@ -86,9 +86,12 @@ if [ ${obsprep_lghtn} -eq 1 ] ; then
-${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
+#${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
+
+cpreq -p ${COMINPREP}/${NET}.${YYYYMMDD}/${NET}.t${HH}z.lghtng.tm00.bufr_d ./${NET}.t${cyc}z.lghtng.tm00.bufr_d
```
Shouldn't this be {NET}.t${cyc}z?

MatthewMorris-NOAA commented on scripts/rtma3d/exrtma3d_obsprep_lghtn.ksh:
```diff
@@ -86,9 +86,12 @@ if [ ${obsprep_lghtn} -eq 1 ] ; then
-${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
+#${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
+
+cpreq -p ${COMINPREP}/${NET}.${YYYYMMDD}/${NET}.t${HH}z.lghtng.tm00.bufr_d ./${NET}.t${cyc}z.lghtng.tm00.bufr_d
```
Shouldn't this be "cyc" instead of "HH"?

EdwardColon-NOAA committed:
Update exrtma3d_obsprep_lghtn.ksh

EdwardColon-NOAA committed:
Update exakrtma3d_obsprep_lghtn.ksh

EdwardColon-NOAA commented:
Hi Annette,

Would you mind checking the following directory to insure that the correct
mrms grib2 files are being archived? Thank you.

/lfs/h2/emc/ptmp/edward.colon/RTMA3D/conus/com2/rtma3d/pbspro/rtma3d.20250507/obsprd.t12z

-Edward

On Wed, May 7, 2025 at 10:26 AM MatthewMorris-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In scripts/rtma3d/exrtma3d_obsprep_lghtn.ksh
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#discussion_r2077764081>:
>
> >
> -${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
> +#${LN} -sf ${NET}.t${cyc}z.lghtng.tm00.bufr_d  lghtngbufr
> +
> +cpreq -p ${COMINPREP}/${NET}.${YYYYMMDD}/${NET}.t${HH}z.lghtng.tm00.bufr_d ./${NET}.t${cyc}z.lghtng.tm00.bufr_d
>
> Shouldn't this be "cyc" instead of "HH"?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#pullrequestreview-2821980246>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQEVROYOKURPYZWTBID25IJ23AVCNFSM6AAAAAB4QDOY6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMRRHE4DAMRUGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

AnnetteGibbs-NOAA commented:
Looks good Ed.  Thanks for making this change.  Have you applied it to the Alaska scripts as well?

EdwardColon-NOAA committed:
Update exakrtma3d_obsprep_radar.ksh

EdwardColon-NOAA committed:
Update exrtma3d_obsprep_radar.ksh

EdwardColon-NOAA commented:
Hi Annette,

I updated the conus and alaska radar processing exscripts. I think the
suggested changes are complete in the PR.

-Edward


On Wed, May 7, 2025 at 12:33 PM Annette Gibbs ***@***.***>
wrote:

> *AnnetteGibbs-NOAA* left a comment (NOAA-EMC/NOAA_3drtma#62)
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#issuecomment-2859247615>
>
> Looks good Ed. Thanks for making this change. Have you applied it to the
> Alaska scripts as well?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/62#issuecomment-2859247615>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQFVZ64BWZE72XNQGBL25IYURAVCNFSM6AAAAAB4QDOY6KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNJZGI2DONRRGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

AnnetteGibbs-NOAA reviewed (approved):
Thanks for making the changes Ed.  I approve.
